doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = iachar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Index function
+
+Position of a *substring* within a *string*.
+
+#### Description
+
+Returns the position of the start of the leftmost or rightmost occurrence
+of string *substring* in *string*, counting from one. If *substring* is not
+present in *string*, zero is returned.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):index(interface)]] (string, substring[, back])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Either scalar character value or string type. This argument is `intent(in)`.
+- `substring`: Either scalar character value or string type. This argument is `intent(in)`.
+- `back`: Either absent or a scalar logical value. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: pos
+
+string = "Search this string for this expression"
+pos = index(string, "this")
+! pos == 8
+
+pos = index(string, "this", back=.true.)
+! pos == 24
+
+pos = index(string, "This")
+! pos == 0
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Scan function
+
+Scan a *string* for the presence of a *set* of characters. Scans a *string* for
+any of the characters in a *set* of characters.
+
+#### Description
+
+If *back* is either absent or *false*, this function returns the position
+of the leftmost character of *string* that is in *set*. If *back* is *true*,
+the rightmost position is returned. If no character of *set* is found in
+*string*, the result is zero.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):scan(interface)]] (string, set[, back])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Either scalar character value or string type. This argument is `intent(in)`.
+- `set`: Either scalar character value or string type. This argument is `intent(in)`.
+- `back`: Either absent or a scalar logical value. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: pos
+
+string = "fortran"
+pos = scan(string, "ao")
+! pos == 2
+
+pos = scan(string, "ao", .true.)
+! pos == 6
+
+pos = scan(string, "c++")
+! pos == 0
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Verify function
+
+Scan a string for the absence of a set of characters. Verifies that all
+the characters in string belong to the set of characters in set.
```

Comment:
The scan function used italic for string and set variable names. I think it should be here too for consistency or?